compiler/rustc_resolve/src/ident.rs

```diff
@@ -1200,7 +1199,6 @@ impl<'a, 'tcx> Resolver<'a, 'tcx> {
                                             }
                                         };
                                         self.report_error(span, error);
```

Comment:
```suggestion
                                        let _: ErrorGuaranteed = self.report_error(span, error);
```
it's non-trivial that `report_error` actually always reports an error so it would be good to propagate this outwards, iirc this code previously had a mode where it silently failed :thinking: apparently that has changed since then

Reply:
I've made the suggested change. It would be nice if `Res::Err` could contain an `ErrorGuaranteed`, but there are many places where one is constructed without a guarantee being available.